godot-core/src/engine/gfile.rs

```diff
@@ -0,0 +1,823 @@
+        })?;
+
+        Ok(Self::from_inner(fa))
+    }
+
+    /// Open a compressed file
+    ///
+    /// Opens a compressed file located at `path`, creating new [`GFile`] object.
+    /// Can read only files compressed by Godot compression formats.
+    /// For [`ModeFlags`] description check the [`GFile`] documentation.
+    pub fn open_compressed(
+        path: impl Into<GString>,
+        flags: ModeFlags,
+        compression_mode: CompressionMode,
+    ) -> IoResult<Self> {
+        let path: GString = path.into();
+        let fa = FileAccess::open_compressed_ex(path.clone(), flags)
+            .compression_mode(compression_mode)
+            .done()
+            .ok_or_else(|| {
+                IoError::new(
+                    ErrorKind::Other,
+                    format!(
+                        "can't open file {} in mode {:?}; GodotError: {:?}",
+                        &path,
+                        flags,
+                        FileAccess::get_open_error()
+                    ),
+                )
+            })?;
+
+        Ok(Self::from_inner(fa))
+    }
+
+    /// Open a file encrypted by byte key
+    ///
+    /// Opens a file encrypted by 32-byte long [`PackedByteArray`] located at
+    /// `path`, creating new [`GFile`] object.
+    /// For [`ModeFlags`] description check the [`GFile`] documentation.
+    pub fn open_encrypted(
+        path: impl Into<GString>,
+        flags: ModeFlags,
+        key: PackedByteArray,
+    ) -> IoResult<Self> {
+        let path: GString = path.into();
+        let fa = FileAccess::open_encrypted(path.clone(), flags, key).ok_or_else(|| {
+            IoError::new(
+                ErrorKind::Other,
+                format!(
+                    "can't open file {} in mode {:?}; GodotError: {:?}",
+                    &path,
+                    flags,
+                    FileAccess::get_open_error()
+                ),
+            )
+        })?;
+
+        Ok(Self::from_inner(fa))
+    }
+
+    /// Open a file encrypted by password
+    ///
+    /// Opens a file encrypted by a `pass` located at `path`, creating new
+    /// [`GFile`] object. For [`ModeFlags`] description check the [`GFile`]
+    /// documentation.
+    pub fn open_encrypted_with_pass(
+        path: impl Into<GString>,
+        flags: ModeFlags,
+        pass: GString,
+    ) -> IoResult<Self> {
+        let path: GString = path.into();
+        let fa =
+            FileAccess::open_encrypted_with_pass(path.clone(), flags, pass).ok_or_else(|| {
+                IoError::new(
+                    ErrorKind::Other,
+                    format!(
+                        "can't open file {} in mode {:?}; GodotError: {:?}",
+                        &path,
+                        flags,
+                        FileAccess::get_open_error()
+                    ),
+                )
+            })?;
+        Ok(Self::from_inner(fa))
+    }
+
+    /// Creates new [`GFile`] from provided [`Gd`]<[`FileAccess`]>.
```

Comment:
Doc link doesn't work, also elsewhere. There's no need to repeat the types, users can click on them in the parameter list.

I'd rather have a brief description such as:
```suggestion
    /// Creates new [`GFile`] from a `FileAccess` pointer with a reference count of 1.
```

Also, name should be `try_from_unique`, in case we later want to add a `from_unique` that panics.